lib/trino-parquet/src/test/java/io/trino/parquet/TestTupleDomainParquetPredicate.java

```diff
@@ -605,6 +606,39 @@ public void testVarcharMatchesWithDictionaryDescriptor()
         assertTrue(parquetPredicate.matches(new DictionaryDescriptor(column, Optional.of(page))));
     }
 
+    @Test
+    public void testEmptyDictionary()
+    {
+        ColumnDescriptor columnDescriptor = new ColumnDescriptor(new String[] {"path"}, BINARY, 0, 0);
+        RichColumnDescriptor column = new RichColumnDescriptor(columnDescriptor, new PrimitiveType(OPTIONAL, BINARY, "Test column"));
+        ColumnDescriptor descriptor = new ColumnDescriptor(column.getPath(), column.getPrimitiveType().getPrimitiveTypeName(), 0, 0);
+        VarcharType type = createVarcharType(255);
+
+        DictionaryPage dictionary = new DictionaryPage(Slices.wrappedBuffer(new byte[] {}), 0, PLAIN_DICTIONARY);
+        TupleDomainParquetPredicate predicate;
+
+        // only non-nulls allowed
+        predicate = new TupleDomainParquetPredicate(
+                withColumnDomains(singletonMap(descriptor, singleValue(type, EMPTY_SLICE, false))),
```

Comment:
For only non-nulls, should the Domain be `notNull(type)` instead of `singleValue(type, EMPTY_SLICE, false)` ? The latter seems to be one empty string. Or maybe the code comment needs an update to say `Only a non-null value allowed`